book-content/chapters/04-essential-types-and-annotations.md

```diff
@@ -0,0 +1,2366 @@
+Now we've covered most of the why of TypeScript, it's time to start with the how. We'll cover key concepts like type annotations and type inference, as well as how to start writing type-safe functions.
+
+It's important to build a solid foundation, as everything you'll learn later builds upon what you'll learn in this chapter.
+
+## Basic Annotations
+
+One of the most common things you'll need to do as a TypeScript developer is to annotate your code. Annotations tell TypeScript what type something is supposed to be.
+
+Annotations will often use a `:` - this is used to tell TypeScript that a variable or function parameter is of a certain type.
+
+### Function Parameter Annotations
+
+One of the most important annotations you'll use is for function parameters.
+
+For example, here is a `logAlbumInfo` function that takes in a `title` string, a `trackCount` number, and an `isReleased` boolean:
+
+```ts twoslash
+const logAlbumInfo = (
```

Comment:
using function declaration for this example (and the following ones) could be more readable for some